src/client/sandbox/xhr.js

```diff
@@ -103,55 +124,24 @@ export default class XhrSandbox extends SandboxBase {
 
             xhrSandbox.emit(xhrSandbox.BEFORE_XHR_SEND_EVENT, { xhr });
 
-            const orscHandler = () => {
-                if (this.readyState === 4)
-                    xhrSandbox.emit(xhrSandbox.XHR_COMPLETED_EVENT, { xhr });
-            };
-
-            // NOTE: If we're using the sync mode or if the response is in cache,
-            // we need to raise the callback manually.
-            if (this.readyState === 4)
-                orscHandler();
-            else {
-                // NOTE: Get out of the current execution tick and then proxy onreadystatechange,
-                // because jQuery assigns a handler after the send() method was called.
-                nativeMethods.setTimeout.call(xhrSandbox.window, () => {
-                    // NOTE: If the state is already changed, we just call the handler without proxying
-                    // onreadystatechange.
-                    if (this.readyState === 4)
-                        orscHandler();
-
-                    else if (typeof this.onreadystatechange === 'function') {
-                        const originalHandler = this.onreadystatechange;
-
-                        this.onreadystatechange = progress => {
-                            orscHandler();
-                            originalHandler.call(this, progress);
-                        };
-                    }
-                    else
-                        this.addEventListener('readystatechange', orscHandler, false);
-                }, 0);
-            }
-
             // NOTE: Add the XHR request mark, so that a proxy can recognize a request as a XHR request. As all
             // requests are passed to the proxy, we need to perform Same Origin Policy compliance checks on the
             // server side. So, we pass the CORS support flag to inform the proxy that it can analyze the
             // Access-Control_Allow_Origin flag and skip "preflight" requests.
-            nativeMethods.xhrSetRequestHeader.call(this, XHR_HEADERS.requestMarker, 'true');
-
-            nativeMethods.xhrSetRequestHeader.call(this, XHR_HEADERS.origin, getOriginHeader());
+            nativeMethods.xhrSetRequestHeader.call(xhr, XHR_HEADERS.requestMarker, 'true');
+            nativeMethods.xhrSetRequestHeader.call(xhr, XHR_HEADERS.origin, getOriginHeader());
 
             if (xhrSandbox.corsSupported)
-                nativeMethods.xhrSetRequestHeader.call(this, XHR_HEADERS.corsSupported, 'true');
+                nativeMethods.xhrSetRequestHeader.call(xhr, XHR_HEADERS.corsSupported, 'true');
```

Comment:
We don't need the additional variable `xhr`.
Another methods (`abort`, `open`) don't declare an additional variable.
So, drop `const xhr = this;` line and change `xhr` to `this` for places in `send` method.